src/host/main.cc

```diff
@@ -1,52 +1,248 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <GLFW/glfw3.h>
+
 #include "burner.h"
 
 // Linux headers
 #include <dlfcn.h>
+#include <time.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <errno.h>
+#include <string.h>
 
 #define TITLE "burner"
 
+#if defined(DEVELOPMENT_MODE)
+    #warning "DEV"
+
+	#define call_start(game__, state__, renderer__, user_data__) (* game__.start)(state__, renderer__, user_data__)
+	#define call_update(game__, state__, renderer__, input__, time__, delta__, user_data__) (* game__.update)(state__, renderer__, input__, time__, delta__, user_data__)
+	#define call_on_reload(game__, state__, renderer__, user_data__) (* game__.on_reload)(state__, renderer__, user_data__)
+#else
+    #warning "RELEASE"
+
+	#define call_start(game__, state__, renderer__, user_data__) start(state__, renderer__, user_data__)
+	#define call_update(game__, state__, renderer__, input__, time__, delta__, user_data__) update(state__, renderer__, input__, time__, delta__, user_data__)
+	#define call_on_reload(game__, state__, renderer__, user_data__)
+
+#endif
+
 void window_error(int error, const char *message)
 {
 	fprintf(stderr, "Window error: %s\n", message);
 }
 
-int main()
+struct FileConfig {
+	const char * src_path;
+	const char * lib_name;
+	const char * shaders_path;
+};
+
+struct GameLibrary {
+	void *lib;
+	game_start *start;
+	game_update *update;
+	game_on_reload *on_reload;
+
+	time_t prev_load_time;
+
+	FileConfig conf;
+};
+
+bool file_modification_time_try_to_update(const char * const file_path, time_t *time_to_update)
 {
-	/* Load game function pointers */
+	struct stat file_info;
+	if (stat(file_path, &file_info)) {
+		fprintf(stderr, "Error getting [%s] source modification time: %s\n", 
+			file_path, strerror(errno));
+		return false;
+	}
+	if (file_info.st_mtime > *time_to_update) {
+		*time_to_update = file_info.st_mtime;
+		return true;
+	}
+	
+	return false;
+}
+
+static const char * const swap_names[2] = {
```

Comment:
Last time I checked, something breaks if I try recompiling a dynamic library while it's in use, so I swap between two library names. I'll re-verify this.